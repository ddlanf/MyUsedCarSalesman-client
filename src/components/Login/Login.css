.login-box{
    margin-top: 12%;
    margin-right: auto;
    margin-left: auto;
    margin-bottom: 10vw;
    width: 90%;
    max-width: 500px;
    text-align: center;
    border: black solid 2px;
}

.login-box > *{
    font-size: 5vw;
}

.login-header-box{
    text-align: center;
    position: relative;
    margin-bottom: 2%;
}

.login-header{
    display: inline;
    font-size: 10vw;
}


.login-label, .login-input{
    display: block;
    margin-left: auto;
    margin-right: auto;
    margin-top: 0;
}

.login-input{
    margin-bottom: 5%;
    margin-top: 0;
    height: 5vw;
    max-height: 55px;
    width: 70%;
    font-size: 2.5vw;
}

.login-label{
    margin-bottom: 2%;
    font-size: 5vw;
}
  
.login-submit{
    margin: 2% auto;
    width: 35%;
    height: 8vw;
    max-height: 55px;
    vertical-align: middle;
    font-size: 3vw;
}

.login-link{
    color: black;
    text-decoration: none;
}

.login-error{
    color: red
}
  
@media screen and (min-width: 1000px) {
  
      .login-box > *{
        font-size: 30px;
      }
  
      .login-header{
        font-size: 55px;
      }

      .login-label, .login-input{
        font-size: 30px;
      }
}