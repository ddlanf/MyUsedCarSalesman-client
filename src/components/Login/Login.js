import React, { Component } from 'react'
import './Login.css'
import {Link} from 'react-router-dom'
import AuthApiService from '../../services/auth-api-service'
import PostContext from '../../contexts/PostContext'

export default class Login extends Component {

    static contextType = PostContext

    constructor(props){
        super(props);
        this.state = { 
            user_name: '',
            password: '',
            error: ''
        }
    }

    loginUser = (ev) =>{
        ev.preventDefault()
        const { user_name, password } = this.state
        const returningUser = { user_name, password }

        let validUser = true
        for(let [key, value] of Object.entries(returningUser)){
            if(value == null || value === ''){
               this.context.setError(`please enter ${key.replace('_', ' ')}`)
               validUser = false
            }
        }

        if(validUser){
            this.context.clearError()
            AuthApiService.postLogin(returningUser)
                .then(user => {
                    this.props.history.push('/view-posts')
                    this.props.userLogIn()
                })
                .catch(res => {
                    this.context.setError(res.error)
                })
        }
    }

    handleInputChange = (event) =>{
        const { name } = event.target
        const { value } = event.target
        
        this.setState({
            [name] : value
        })
    }


    render() {
        return (
                <div className="login-box">
                    <div className="login-header-box">
                     <h1 className="login-header">Log in</h1>
                    </div>
                    <form
                        onSubmit={this.loginUser}
                        >
                        <label name="user-name" className="login-label">User Name/Email</label>
                        <input 
                            type="text" 
                            className="login-input"
                            name="user_name"
                            onChange={this.handleInputChange}/>
                        <label name="user-password" className="login-label">Password</label>
                        <input 
                            type="password" 
                            className="login-input"
                            name="password"
                            onChange={this.handleInputChange}/>
                        <input type="submit" className="login-submit"></input>
                    </form>
                    {<p className='login-error'>{this.context.error}</p>}
                    <p>New to MyUsedCarSalesman?
                        <span className="break">
                            <Link
                                className="login-link"
                                to='/register'
                                >
                                Register your account
                            </Link>
                       </span>
                    </p>
                    <p>
                        <Link
                            className="login-link"
                            to='/admin-login'
                            >
                         Admin Login
                        </Link>
                    </p>
                </div>
        )
    }
}
