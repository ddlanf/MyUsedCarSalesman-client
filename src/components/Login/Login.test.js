import React from 'react';
import ReactDOM from 'react-dom';
import Login from './Login';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

describe('Login component', () => {

    it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<BrowserRouter>
                       <Login
                          isUserLoggedIn={false} 
                          userLogIn={function(){}}
                          userIsAtLoginOrRegister={function(){}}
                          userIsNotAtLoginOrRegister={function(){}}
                          isUserAtLoginOrRegister={false}
                        />
                    </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
          .create(
                <BrowserRouter>
                    <Login
                         isUserLoggedIn={false} 
                         userLogIn={function(){}}
                         userIsAtLoginOrRegister={function(){}}
                         userIsNotAtLoginOrRegister={function(){}}
                         isUserAtLoginOrRegister={false}
                    />
                </BrowserRouter>)
          .toJSON();
        expect(tree).toMatchSnapshot();  
      });
});