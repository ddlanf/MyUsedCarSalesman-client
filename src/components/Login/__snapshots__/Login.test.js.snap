// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders the UI as expected 1`] = `
<div
  className="login-box"
>
  <div
    className="login-header-box"
  >
    <h1
      className="login-header"
    >
      Log in
    </h1>
  </div>
  
  <form
    onSubmit={[Function]}
  >
    <label
      className="login-label"
      name="user-name"
    >
      User Name
    </label>
    <input
      className="login-input"
      name="user_name"
      onChange={[Function]}
      type="text"
    />
    <label
      className="login-label"
      name="user-password"
    >
      Password
    </label>
    <input
      className="login-input"
      name="password"
      onChange={[Function]}
      type="password"
    />
    <button
      className="login-submit"
      type="submit"
    >
      Submit
    </button>
  </form>
  <h3
    className="login-register-heading"
  >
    New to MyUsedCarSalesman?
    <span
      className="break"
    >
      <a
        className="register-login-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </span>
  </h3>
  <h3
    className="login-register-heading"
  >
    <a
      className="login-register-link"
      href="/admin-login"
      onClick={[Function]}
    >
      Admin Login
    </a>
  </h3>
</div>
`;
